source/common/quic/envoy_quic_server_connection.cc

```diff
@@ -47,6 +71,13 @@ bool EnvoyQuicServerConnection::OnPacketHeader(const quic::QuicPacketHeader& hea
   return true;
 }
 
+void EnvoyQuicServerConnection::OnWritePacketDone(size_t packet_size,
+                                                  const quic::WriteResult& /*result*/) {
+  if (hasConnectionStats()) {
+    connectionStats().write_total_.add(packet_size);
```

Comment:
I think this will count the total number of bytes sent, including retransmissions.  Based on https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/stats#statistics, that seems to be the intended semantics of this counter, but I wasn't 100% sure.

Reply:
I agree it seems subject to interpretation.  Including retransmissions SGTM.

Reply:
IIUC, there is no retransmission at the H2 layer, it just rely on TCP retransmission. The `delta` probably includes H2 framing overhead, which is (obviously) H2-specific.